### ISSUE TITLE
cmake re-runs too often (always!)

### ISSUE THREAD

**Issue by tomjakubowski** [Oblong/obi#38]
Every time I run `obi build`, cmake is re-run and I see a whole lot of spew from FindGSpeak. This re-run isn't necessary; cmake's generated Makefiles will re-run cmake if CMakeLists.txt was changed.

The "configure" task would run cmake (in the default case) to populate the build directory, while the "build" task would run `cmake --build build/ -- $BUILD_FLAGS`(in the default case). `obi configure` could do nothing if it detects that the build directory exists.

**Comment by dankegel** [comment]
Since cmake autoruns if you changed anything since last make, the simplest way to get what you want is to only run cmake explicitly if it's never been run before, innit?  No need for an extra task...?

**Comment by tomjakubowski** [comment]
Yeah, that sounds better.

**Comment by tomjakubowski** [comment]
slightly sanitized example of FindGSpeak spew. I love to see this every time I run `obi build`:

```
% obi build
[localhost] Executing task '<lambda>'
[localhost] Executing task 'build_task'
[localhost] local: mkdir -p /Users/tjakubowski/obsrc/dieter/sprockets/build
[localhost] local: cmake -H"/Users/tjakubowski/obsrc/dieter/sprockets" -B"/Users/tjakubowski/obsrc/dieter/sprockets/build" -DCMAKE_BUILD_TYPE=RelWithDebInfo -DCMAKE_EXPORT_COMPILE_COMMANDS=1 -DSPROCKETS_ENABLE_GROWROOM=1 -DENABLE_REMOTERY=1
-- Setting environment variable PKG_CONFIG_PATH=/opt/oblong/g-speak3.28/lib/pkgconfig:/opt/oblong/deps-64-11/lib/pkgconfig
-- Setting environment variable PATH=/Users/tjakubowski/.local/bin:/Users/tjakubowski/bin:/Users/tjakubowski/gocode/bin:/bin:/Users/tjakubowski/.rbenv/bin:/usr/local/bin:/usr/bin:/usr/sbin:/sbin:/opt/X11/bin:/opt/oblong/g-speak3.23/bin:/opt/oblong/deps-64-10/bin:/opt/oblong/g-speak3.22/bin:/opt/oblong/g-speak3.21/bin:/opt/oblong/g-speak3.20/bin:/opt/oblong/g-speak3.18/bin:/opt/oblong/greenhouse/bin:/opt/oblong/g-speak3.16/bin:/opt/oblong/deps-64-9/bin:/Users/tjakubowski/.cargo/bin:/opt/oblong/g-speak3.28/bin:/opt/oblong/deps-64-11/bin
-- G_SPEAK_LDFLAGS is -L/opt/oblong/g-speak3.28/lib -L/opt/oblong/deps-64-11/lib -L/opt/oblong/g-speak3.28/lib -L/opt/oblong/deps-64-11/lib -L/opt/oblong/g-speak3.28/lib -L/opt/oblong/deps-64-11/lib -L/opt/oblong/g-speak3.28/lib -L/opt/oblong/deps-64-11/lib -L/opt/oblong/g-speak3.28/lib -L/opt/oblong/deps-64-11/lib -L/opt/oblong/deps-64-11/cef2704/Release/lib -L/opt/oblong/deps-64-11/lib -L/opt/oblong/deps-64-11/v8/5.2.361/lib/x64.release.libc++  -lGrowroom  -lGestation  -lPlasmaZeroconf -lyaml -lssl -lcrypto  -lStaging -framework Cocoa -framework ApplicationServices -lboost_filesystem-mt -lboost_regex-mt -lboost_system-mt -lboost_thread-mt  -lAfferent  -lTwillig  -lWebThing -F/opt/oblong/deps-64-11/cef2704/Release/cefclient.app/Contents/Frameworks -framework 'Chromium Embedded Framework' -lcef_dll_wrapper  -lNoodoo -framework OpenGL -framework AGL -lfontconfig -lfreetype -lm -lpthread  -ljpeg -ltiff -lpng12 -framework Cocoa  -lImpetus  -lMedia -lm -lpthread  -ljpeg -ltiff -lpng12 -lgnutls  -lplasma-gstreamer -lgstreamer-1.0 -lgobject-2.0 -lglib-2.0 -lintl -lm -lpthread  -ljpeg -ltiff -lpng12  -lBasement -lgstvideo-1.0 -lgstapp-1.0 -lgstbase-1.0 -lgstrtsp-1.0 -lgstreamer-1.0 -lgstsdp-1.0 -lgio-2.0 -lgobject-2.0 -lglib-2.0 -lintl  -lPlasma++  -lLoam++ -lboost_filesystem-mt -lboost_regex-mt -lboost_system-mt -lboost_thread-mt -licuuc -licui18n  -lPlasma -lyaml -lssl -lcrypto  -lLoam -lm -lpthread  -ljpeg -ltiff -lpng12 -lGreenhouse -lGestation -framework OpenGL -framework AGL -framework Cocoa -framework OpenGL -framework AGL -framework Cocoa -lPlasmaZeroconf -lStaging -framework Cocoa -framework ApplicationServices -framework OpenGL -framework AGL -framework Cocoa -framework OpenGL -framework AGL -framework Cocoa -lAfferent -lTwillig -lNoodoo -framework OpenGL -framework AGL -framework Cocoa -lImpetus -lMedia -lgnutls -lplasma-gstreamer -lBasement -lgstvideo-1.0 -lgstapp-1.0 -lgstbase-1.0 -lgstrtsp-1.0 -lgstreamer-1.0 -lgstsdp-1.0 -lgio-2.0 -lgobject-2.0 -lglib-2.0 -lintl -lPlasma++ -lLoam++ -lboost_filesystem-mt -lboost_regex-mt -lboost_system-mt -lboost_thread-mt -licuuc -licui18n -lPlasma -lyaml -lssl -lcrypto -lLoam -lm -lpthread -ljpeg -ltiff -lpng12 -lfontconfig -lfreetype  -lv8 -lv8_libbase -lv8_libplatform
-- G_SPEAK_CFLAGS is -D_REENTRANT -pthread -D_REENTRANT -pthread -D_REENTRANT -pthread -D_REENTRANT -pthread -isystem /opt/oblong/g-speak3.28/include -isystem /opt/oblong/deps-64-11/include -isystem /opt/oblong/deps-64-11/include/gstreamer-1.0 -isystem /opt/oblong/deps-64-11/lib/gstreamer-1.0/include -isystem /opt/oblong/deps-64-11/include/glib-2.0 -isystem /opt/oblong/deps-64-11/lib/glib-2.0/include -isystem /opt/oblong/deps-64-11/cef2704/include -isystem /opt/oblong/g-speak3.28/include/libGreenhouse -isystem /opt/oblong/deps-64-11/include/freetype2 -isystem /opt/oblong/deps-64-11/v8/5.2.361/include -isystem /opt/oblong/deps-64-11/v8/5.2.361/include/.. -isystem /opt/oblong/g-speak3.28/include -isystem /opt/oblong/deps-64-11/include -isystem /opt/oblong/deps-64-11/include/gstreamer-1.0 -isystem /opt/oblong/deps-64-11/lib/gstreamer-1.0/include -isystem /opt/oblong/deps-64-11/include/glib-2.0 -isystem /opt/oblong/deps-64-11/lib/glib-2.0/include -isystem /opt/oblong/g-speak3.28/include -isystem /opt/oblong/deps-64-11/include -isystem /opt/oblong/g-speak3.28/include -isystem /opt/oblong/deps-64-11/include -isystem /opt/oblong/g-speak3.28/include/libGreenhouse -isystem /opt/oblong/g-speak3.28/include -isystem /opt/oblong/deps-64-11/include -isystem /opt/oblong/deps-64-11/include/gstreamer-1.0 -isystem /opt/oblong/deps-64-11/lib/gstreamer-1.0/include -isystem /opt/oblong/deps-64-11/include/glib-2.0 -isystem /opt/oblong/deps-64-11/lib/glib-2.0/include -isystem /opt/oblong/deps-64-11/include/freetype2 -isystem /opt/oblong/deps-64-11/cef2704/include -isystem /opt/oblong/deps-64-11/include -isystem /opt/oblong/deps-64-11/include/gstreamer-1.0 -isystem /opt/oblong/deps-64-11/lib/gstreamer-1.0/include -isystem /opt/oblong/deps-64-11/include/gstreamer-1.0 -isystem /opt/oblong/deps-64-11/include/glib-2.0 -isystem /opt/oblong/deps-64-11/lib/glib-2.0/include -isystem /opt/oblong/deps-64-11/v8/5.2.361/include -isystem /opt/oblong/deps-64-11/v8/5.2.361/include/..
-- G_SPEAK_INCLUDE_DIRS is /opt/oblong/g-speak3.28/include;/opt/oblong/deps-64-11/include;/opt/oblong/deps-64-11/include/gstreamer-1.0;/opt/oblong/deps-64-11/lib/gstreamer-1.0/include;/opt/oblong/deps-64-11/include/glib-2.0;/opt/oblong/deps-64-11/lib/glib-2.0/include;/opt/oblong/deps-64-11/cef2704/include;/opt/oblong/g-speak3.28/include/libGreenhouse;/opt/oblong/deps-64-11/include/freetype2;/opt/oblong/deps-64-11/v8/5.2.361/include;/opt/oblong/deps-64-11/v8/5.2.361/include/..;/opt/oblong/g-speak3.28/include;/opt/oblong/deps-64-11/include;/opt/oblong/deps-64-11/include/gstreamer-1.0;/opt/oblong/deps-64-11/lib/gstreamer-1.0/include;/opt/oblong/deps-64-11/include/glib-2.0;/opt/oblong/deps-64-11/lib/glib-2.0/include;/opt/oblong/g-speak3.28/include;/opt/oblong/deps-64-11/include;/opt/oblong/g-speak3.28/include;/opt/oblong/deps-64-11/include;/opt/oblong/g-speak3.28/include/libGreenhouse;/opt/oblong/g-speak3.28/include;/opt/oblong/deps-64-11/include;/opt/oblong/deps-64-11/include/gstreamer-1.0;/opt/oblong/deps-64-11/lib/gstreamer-1.0/include;/opt/oblong/deps-64-11/include/glib-2.0;/opt/oblong/deps-64-11/lib/glib-2.0/include;/opt/oblong/deps-64-11/include/freetype2;/opt/oblong/deps-64-11/cef2704/include;/opt/oblong/deps-64-11/include;/opt/oblong/deps-64-11/include/gstreamer-1.0;/opt/oblong/deps-64-11/lib/gstreamer-1.0/include;/opt/oblong/deps-64-11/include/gstreamer-1.0;/opt/oblong/deps-64-11/include/glib-2.0;/opt/oblong/deps-64-11/lib/glib-2.0/include;/opt/oblong/deps-64-11/v8/5.2.361/include;/opt/oblong/deps-64-11/v8/5.2.361/include/..
-- G_SPEAK_STATIC_LDFLAGS is -L/opt/oblong/g-speak3.28/lib -L/opt/oblong/deps-64-11/lib -L/opt/oblong/g-speak3.28/lib -L/opt/oblong/deps-64-11/lib -L/opt/oblong/g-speak3.28/lib -L/opt/oblong/deps-64-11/lib -L/opt/oblong/g-speak3.28/lib -L/opt/oblong/deps-64-11/lib -L/opt/oblong/g-speak3.28/lib -L/opt/oblong/deps-64-11/lib -L/opt/oblong/g-speak3.28/lib -L/opt/oblong/deps-64-11/lib -L/opt/oblong/g-speak3.28/lib -L/opt/oblong/deps-64-11/lib -L/opt/oblong/deps-64-11/cef2704/Release/lib -L/opt/oblong/deps-64-11/lib -L/opt/oblong/deps-64-11/v8/5.2.361/lib/x64.release.libc++  -lGrowroom  -lGestation  -lPlasmaZeroconf -lyaml -lssl -lcrypto  -lStaging -framework Cocoa -framework ApplicationServices -lboost_filesystem-mt -lboost_regex-mt -lboost_system-mt -lboost_thread-mt  -lAfferent  -lTwillig  -lWebThing -lGestation -lAfferent -framework OpenGL -framework AGL -framework Cocoa -framework OpenGL -framework AGL -framework Cocoa -lPlasmaZeroconf -lTwillig -lNoodoo -framework OpenGL -framework AGL -lfontconfig -lfreetype -framework Cocoa -lImpetus -lMedia -lgnutls -lplasma-gstreamer -lBasement -lgstvideo-1.0 -lgstapp-1.0 -lgstbase-1.0 -lgstrtsp-1.0 -lgstreamer-1.0 -lgstsdp-1.0 -lgio-2.0 -lgobject-2.0 -lglib-2.0 -lintl -lPlasma++ -lLoam++ -lboost_filesystem-mt -lboost_regex-mt -lboost_system-mt -lboost_thread-mt -licuuc -licui18n -lPlasma -lyaml -lssl -lcrypto -lLoam -lm -lpthread -ljpeg -ltiff -lpng12 -lstdc++ -F/opt/oblong/deps-64-11/cef2704/Release/cefclient.app/Contents/Frameworks -framework 'Chromium Embedded Framework' -lcef_dll_wrapper  -lNoodoo -framework OpenGL -framework AGL -lfontconfig -lfreetype -lm -lpthread  -ljpeg -ltiff -lpng12 -framework Cocoa  -lImpetus  -lMedia -lm -lpthread  -ljpeg -ltiff -lpng12 -lgnutls -lz -lintl -liconv -lc -Wl,-framework -Wl,CoreFoundation -liconv -lgmp -lhogweed -lgmp -lnettle  -lplasma-gstreamer -lgstreamer-1.0 -lgobject-2.0 -lglib-2.0 -lintl -lm -lpthread  -ljpeg -ltiff -lpng12  -lBasement -lgstvideo-1.0 -lgstapp-1.0 -lgstbase-1.0 -lgstrtsp-1.0 -lgstreamer-1.0 -lgmodule-2.0 -lgstsdp-1.0 -lgio-2.0 -lgobject-2.0 -lglib-2.0 -lintl  -lPlasma++  -lLoam++ -lboost_filesystem-mt -lboost_regex-mt -lboost_system-mt -lboost_thread-mt -licuuc -licui18n  -lPlasma -lyaml -lssl -lcrypto  -lLoam -lm -lpthread  -ljpeg -ltiff -lpng12 -lGreenhouse -lGestation -framework OpenGL -framework AGL -framework Cocoa -framework OpenGL -framework AGL -framework Cocoa -lPlasmaZeroconf -lStaging -framework Cocoa -framework ApplicationServices -framework OpenGL -framework AGL -framework Cocoa -framework OpenGL -framework AGL -framework Cocoa -lAfferent -lTwillig -lNoodoo -framework OpenGL -framework AGL -framework Cocoa -lImpetus -lMedia -lgnutls -lplasma-gstreamer -lBasement -lgstvideo-1.0 -lgstapp-1.0 -lgstbase-1.0 -lgstrtsp-1.0 -lgstreamer-1.0 -lgstsdp-1.0 -lgio-2.0 -lgobject-2.0 -lglib-2.0 -lintl -lPlasma++ -lLoam++ -lboost_filesystem-mt -lboost_regex-mt -lboost_system-mt -lboost_thread-mt -licuuc -licui18n -lPlasma -lyaml -lssl -lcrypto -lLoam -lm -lpthread -ljpeg -ltiff -lpng12 -lfontconfig -lfreetype  -lv8 -lv8_libbase -lv8_libplatform -L/opt/oblong/g-speak3.28/lib -L/opt/oblong/deps-64-11/lib -L/opt/oblong/g-speak3.28/lib -L/opt/oblong/deps-64-11/lib -L/opt/oblong/g-speak3.28/lib -L/opt/oblong/deps-64-11/lib -L/opt/oblong/g-speak3.28/lib -L/opt/oblong/deps-64-11/lib -L/opt/oblong/g-speak3.28/lib -L/opt/oblong/deps-64-11/lib -L/opt/oblong/g-speak3.28/lib -L/opt/oblong/deps-64-11/lib -L/opt/oblong/g-speak3.28/lib -L/opt/oblong/deps-64-11/lib -L/opt/oblong/deps-64-11/cef2704/Release/lib -L/opt/oblong/deps-64-11/lib -L/opt/oblong/deps-64-11/v8/5.2.361/lib/x64.release.libc++  -lGrowroom  -lGestation  -lPlasmaZeroconf -lyaml -lssl -lcrypto  -lStaging -framework Cocoa -framework ApplicationServices -lboost_filesystem-mt -lboost_regex-mt -lboost_system-mt -lboost_thread-mt  -lAfferent  -lTwillig  -lWebThing -lGestation -lAfferent -framework OpenGL -framework AGL -framework Cocoa -framework OpenGL -framework AGL -framework Cocoa -lPlasmaZeroconf -lTwillig -lNoodoo -framework OpenGL -framework AGL -lfontconfig -lfreetype -framework Cocoa -lImpetus -lMedia -lgnutls -lplasma-gstreamer -lBasement -lgstvideo-1.0 -lgstapp-1.0 -lgstbase-1.0 -lgstrtsp-1.0 -lgstreamer-1.0 -lgstsdp-1.0 -lgio-2.0 -lgobject-2.0 -lglib-2.0 -lintl -lPlasma++ -lLoam++ -lboost_filesystem-mt -lboost_regex-mt -lboost_system-mt -lboost_thread-mt -licuuc -licui18n -lPlasma -lyaml -lssl -lcrypto -lLoam -lm -lpthread -ljpeg -ltiff -lpng12 -lstdc++ -F/opt/oblong/deps-64-11/cef2704/Release/cefclient.app/Contents/Frameworks -framework 'Chromium Embedded Framework' -lcef_dll_wrapper  -lNoodoo -framework OpenGL -framework AGL -lfontconfig -lfreetype -lm -lpthread  -ljpeg -ltiff -lpng12 -framework Cocoa  -lImpetus  -lMedia -lm -lpthread  -ljpeg -ltiff -lpng12 -lgnutls -lz -lintl -liconv -lc -Wl,-framework -Wl,CoreFoundation -liconv -lgmp -lhogweed -lgmp -lnettle  -lplasma-gstreamer -lgstreamer-1.0 -lgobject-2.0 -lglib-2.0 -lintl -lm -lpthread  -ljpeg -ltiff -lpng12  -lBasement -lgstvideo-1.0 -lgstapp-1.0 -lgstbase-1.0 -lgstrtsp-1.0 -lgstreamer-1.0 -lgmodule-2.0 -lgstsdp-1.0 -lgio-2.0 -lgobject-2.0 -lglib-2.0 -lintl  -lPlasma++  -lLoam++ -lboost_filesystem-mt -lboost_regex-mt -lboost_system-mt -lboost_thread-mt -licuuc -licui18n  -lPlasma -lyaml -lssl -lcrypto  -lLoam -lm -lpthread  -ljpeg -ltiff -lpng12 -lGreenhouse -lGestation -framework OpenGL -framework AGL -framework Cocoa -framework OpenGL -framework AGL -framework Cocoa -lPlasmaZeroconf -lStaging -framework Cocoa -framework ApplicationServices -framework OpenGL -framework AGL -framework Cocoa -framework OpenGL -framework AGL -framework Cocoa -lAfferent -lTwillig -lNoodoo -framework OpenGL -framework AGL -framework Cocoa -lImpetus -lMedia -lgnutls -lplasma-gstreamer -lBasement -lgstvideo-1.0 -lgstapp-1.0 -lgstbase-1.0 -lgstrtsp-1.0 -lgstreamer-1.0 -lgstsdp-1.0 -lgio-2.0 -lgobject-2.0 -lglib-2.0 -lintl -lPlasma++ -lLoam++ -lboost_filesystem-mt -lboost_regex-mt -lboost_system-mt -lboost_thread-mt -licuuc -licui18n -lPlasma -lyaml -lssl -lcrypto -lLoam -lm -lpthread -ljpeg -ltiff -lpng12 -lfontconfig -lfreetype  -lv8 -lv8_libbase -lv8_libplatform
-- G_SPEAK_STATIC_CFLAGS is -D_REENTRANT -pthread -D_REENTRANT -pthread -D_REENTRANT -pthread -D_REENTRANT -pthread -isystem /opt/oblong/g-speak3.28/include -isystem /opt/oblong/deps-64-11/include -isystem /opt/oblong/deps-64-11/include/gstreamer-1.0 -isystem /opt/oblong/deps-64-11/lib/gstreamer-1.0/include -isystem /opt/oblong/deps-64-11/include/glib-2.0 -isystem /opt/oblong/deps-64-11/lib/glib-2.0/include -isystem /opt/oblong/deps-64-11/cef2704/include -isystem /opt/oblong/g-speak3.28/include/libGreenhouse -isystem /opt/oblong/deps-64-11/include/freetype2 -isystem /opt/oblong/deps-64-11/v8/5.2.361/include -isystem /opt/oblong/deps-64-11/v8/5.2.361/include/.. -isystem /opt/oblong/g-speak3.28/include -isystem /opt/oblong/deps-64-11/include -isystem /opt/oblong/deps-64-11/include/gstreamer-1.0 -isystem /opt/oblong/deps-64-11/lib/gstreamer-1.0/include -isystem /opt/oblong/deps-64-11/include/glib-2.0 -isystem /opt/oblong/deps-64-11/lib/glib-2.0/include -isystem /opt/oblong/g-speak3.28/include -isystem /opt/oblong/deps-64-11/include -isystem /opt/oblong/g-speak3.28/include -isystem /opt/oblong/deps-64-11/include -isystem /opt/oblong/g-speak3.28/include/libGreenhouse -isystem /opt/oblong/g-speak3.28/include -isystem /opt/oblong/deps-64-11/include -isystem /opt/oblong/deps-64-11/include/gstreamer-1.0 -isystem /opt/oblong/deps-64-11/lib/gstreamer-1.0/include -isystem /opt/oblong/deps-64-11/include/glib-2.0 -isystem /opt/oblong/deps-64-11/lib/glib-2.0/include -isystem /opt/oblong/deps-64-11/include/freetype2 -isystem /opt/oblong/deps-64-11/cef2704/include -isystem /opt/oblong/deps-64-11/include -isystem /opt/oblong/deps-64-11/include/gstreamer-1.0 -isystem /opt/oblong/deps-64-11/lib/gstreamer-1.0/include -isystem /opt/oblong/deps-64-11/include/gstreamer-1.0 -isystem /opt/oblong/deps-64-11/include/glib-2.0 -isystem /opt/oblong/deps-64-11/lib/glib-2.0/include -isystem /opt/oblong/deps-64-11/v8/5.2.361/include -isystem /opt/oblong/deps-64-11/v8/5.2.361/include/..
-- G_SPEAK_STATIC_INCLUDE_DIRS is /opt/oblong/g-speak3.28/include;/opt/oblong/deps-64-11/include;/opt/oblong/deps-64-11/include/gstreamer-1.0;/opt/oblong/deps-64-11/lib/gstreamer-1.0/include;/opt/oblong/deps-64-11/include/glib-2.0;/opt/oblong/deps-64-11/lib/glib-2.0/include;/opt/oblong/deps-64-11/cef2704/include;/opt/oblong/g-speak3.28/include/libGreenhouse;/opt/oblong/deps-64-11/include/freetype2;/opt/oblong/deps-64-11/v8/5.2.361/include;/opt/oblong/deps-64-11/v8/5.2.361/include/..;/opt/oblong/g-speak3.28/include;/opt/oblong/deps-64-11/include;/opt/oblong/deps-64-11/include/gstreamer-1.0;/opt/oblong/deps-64-11/lib/gstreamer-1.0/include;/opt/oblong/deps-64-11/include/glib-2.0;/opt/oblong/deps-64-11/lib/glib-2.0/include;/opt/oblong/g-speak3.28/include;/opt/oblong/deps-64-11/include;/opt/oblong/g-speak3.28/include;/opt/oblong/deps-64-11/include;/opt/oblong/g-speak3.28/include/libGreenhouse;/opt/oblong/g-speak3.28/include;/opt/oblong/deps-64-11/include;/opt/oblong/deps-64-11/include/gstreamer-1.0;/opt/oblong/deps-64-11/lib/gstreamer-1.0/include;/opt/oblong/deps-64-11/include/glib-2.0;/opt/oblong/deps-64-11/lib/glib-2.0/include;/opt/oblong/deps-64-11/include/freetype2;/opt/oblong/deps-64-11/cef2704/include;/opt/oblong/deps-64-11/include;/opt/oblong/deps-64-11/include/gstreamer-1.0;/opt/oblong/deps-64-11/lib/gstreamer-1.0/include;/opt/oblong/deps-64-11/include/gstreamer-1.0;/opt/oblong/deps-64-11/include/glib-2.0;/opt/oblong/deps-64-11/lib/glib-2.0/include;/opt/oblong/deps-64-11/v8/5.2.361/include;/opt/oblong/deps-64-11/v8/5.2.361/include/..
Seabed is skipping libWebThing
-- Seabed standalone build? 0
-- Building with Remotery profiling support enabled
-- ObRemotery standalone build? 0
-- Configuring done
-- Generating done
-- Build files have been written to: /Users/tjakubowski/obsrc/dieter/sprockets/build
```

So I think this is still the question to be answered: how do we cleanly accomplish this?

> the simplest way to get what you want is to only run cmake explicitly if it's never been run before

07e290bd4e7e4418156e8f89ba02fb2af48c5fd5 interpreted "existence of the build directory" as sufficient evidence that cmake had been run to completion at least once. But the build directory still exists if cmake stops mid-way due to an error. So we need some stronger medicine.

**Comment by tomjakubowski** [comment]
I did a little experiment with a minimal CMakeLists.txt:

```
project(aaaaaaa)

#message(FATAL_ERROR "oops!")
```

By commenting/uncommenting that line, I can make CMake fail or succeed.

I first did a successful run:

```
mkdir success
cd success
cmake ..
```

And then uncommented that line, and did a failed/partial cmake run:

```
mkdir fail
cd fail
cmake ..
```

Here are the directory trees for each case:

```
success
├── CMakeCache.txt
├── CMakeFiles
│   ├── 3.8.0
│   │   ├── CMakeCCompiler.cmake
│   │   ├── CMakeCXXCompiler.cmake
│   │   ├── CMakeDetermineCompilerABI_C.bin
│   │   ├── CMakeDetermineCompilerABI_CXX.bin
│   │   ├── CMakeSystem.cmake
│   │   ├── CompilerIdC
│   │   │   ├── CMakeCCompilerId.c
│   │   │   ├── a.out
│   │   │   └── tmp
│   │   └── CompilerIdCXX
│   │       ├── CMakeCXXCompilerId.cpp
│   │       ├── a.out
│   │       └── tmp
│   ├── CMakeOutput.log
│   ├── CMakeTmp
│   ├── TargetDirectories.txt
│   ├── cmake.check_cache
│   ├── feature_tests.bin
│   ├── feature_tests.c
│   └── feature_tests.cxx
├── build.ninja
├── cmake_install.cmake
└── rules.ninja

7 directories, 19 files

fail
├── CMakeCache.txt
└── CMakeFiles
    ├── 3.8.0
    │   ├── CMakeCCompiler.cmake
    │   ├── CMakeCXXCompiler.cmake
    │   ├── CMakeDetermineCompilerABI_C.bin
    │   ├── CMakeDetermineCompilerABI_CXX.bin
    │   ├── CMakeSystem.cmake
    │   ├── CompilerIdC
    │   │   ├── CMakeCCompilerId.c
    │   │   ├── a.out
    │   │   └── tmp
    │   └── CompilerIdCXX
    │       ├── CMakeCXXCompilerId.cpp
    │       ├── a.out
    │       └── tmp
    ├── CMakeOutput.log
    ├── CMakeTmp
    ├── cmake.check_cache
    ├── feature_tests.bin
    ├── feature_tests.c
    └── feature_tests.cxx
```

The big differences that pop out, which obi could potentially test for, are:

* presence of `build.ninja` and related in the success case.
* presence of `cmake_install.cmake` in the success case (but probably only if `install()` is called in the CMakeLists.tt)
* presence of a bunch of files under `build/CMakeFiles/3.8.0`

The first one could be good to test for, but we'd also want to check for Makefiles + XCode stuff. 

On the other hand, since we're supposed to support other metabuild systems, obi might be a little too flexible for this feature… should obi cut out support for anything but CMake? @jshrake

**Comment by tomjakubowski** [comment]
Suggestion:

* When obi runs `cmake` (or equivalent) it should check the resulting exit code. If the exit code is `0`, it should touch a file in the build dir with a name like `obi-cmake-success`.
* obi should use the presence/absence of this file to decide whether to run `cmake` again in `build`: if it's there, don't run cmake; if it's not, run cmake.

**Comment by Gobiner** [comment]
Fixed in PR #114
